lib/travis/api/app/endpoint/lint.rb

```diff
@@ -1,19 +1,25 @@
 require 'travis/api/app'
-require 'travis/yaml'
 
 class Travis::Api::App
   class Endpoint
     class Lint < Endpoint
+      WARNING = /\[(alert|error|warn)\]/
+
+      post('/', scope: :public) { lint }
+      put('/',  scope: :public) { lint }
+
       def lint
         request.body.rewind
-        content  = params[:content] || request.body.read
-        parsed   = Travis::Yaml.parse(content)
-        warnings = parsed.nested_warnings.map { |k, m| { key: k, message: m } }
-        { lint: { warnings: warnings } }.to_json
+        content = params[:content] || request.body.read
+        query = Travis::API::V3::Queries::Lint.new({}, :lint)
+        msgs = query.lint(content)
+        { lint: { warnings: warnings(msgs) } }.to_json
```

Comment:
Any reason to duplicate the `warnings` method from `Travis::API::V3::Renderer::Lint` instead of using the module itself?

```suggestion
        warnings = Travis::API::V3::Renderer::Lint.warnings(msgs)
        { lint: { warnings: warnings } }.to_json
```

That'd also prevent duplicating the `WARNING` constant as well.

Reply:
yeah, hmmmm, good question. i suppose in theory `v2` and `v3` code shouldn't overlap at all. `v2` should be retired at some point anyway ... but on top, i wouldn't know how to instantiate the renderer class and make it return the right stuff, i think it might be coupled too tightly with the `v3` to be worth it. if you feel it's worth looking into sharing the method, go for it.